spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -71,7 +74,17 @@ public boolean remove(String path) throws IOException {
 
 	@Override
 	public FTPFile[] list(@Nullable String path) throws IOException {
-		return this.client.listFiles(path);
+		FTPFile[] ftpFiles = this.client.listFiles(path);
+		if (StringUtils.hasText(path)) {
+			List<EnhancedFTPFile> enhancedFtpFiles = new LinkedList<>();
+			for (FTPFile ftpFile : ftpFiles) {
+				EnhancedFTPFile enhancedFTPFile = new EnhancedFTPFile(ftpFile);
+				enhancedFTPFile.setLongFileName(path + '/' + ftpFile.getName());
```

Comment:
If path is the Current Working Directory of "" wouldn't we still want to go through the flow of adding the file name?